src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java

```diff
@@ -73,13 +73,37 @@ public DatastoreScheduledInterviewDao() {
     return scheduledInterviews;
   }
 
+  /**
+   * Returns a list, sorted by start time, of all ScheduledInterview objects ranging from minTime to
+   * maxTime that are for the selected position and do not already have a shadow.
+   */
+  public List<ScheduledInterview> getForPositionWithoutShadowInRange(
+      Job position, Instant minTime, Instant maxTime) {
+    List<Filter> allFilters = new ArrayList<>();
```

Comment:
Kind of a nit: You can further minimize the logic in getPossibleInterviews() in the Load servlet by adding more filters and a shadowId param to this function. Use notEqual filters to perform this logic:
!(interviewerId == userId) && !(intervieweeId == userId) && !(shadow == userId)

Reply:
I tried it, it doesn't work because you're not allowed to have NOT_EQUAL filters on several parameters within the same Query. Definitely good thought though.